src/main.rs

```diff
@@ -1,91 +1,102 @@
+        process::exit(1);
+    }
+}
 
 fn main() {
-    let args: Vec<String> = env::args().collect();
-    let default = "tmp/ipc.sock".to_owned();
-    let ipc_path = args.get(0).unwrap_or(&default);
+    let host = env::args().nth(1).unwrap_or_else(||
+        DFLT_BROKER.to_string()
+    );
 
-    let mut down_command = Command::new("sudo")
-        .arg("ifconfig")
-        .arg("can0")
-        .arg("down")
-        .spawn()
-        .expect("down command did not work");
-    down_command
-        .wait()
-        .expect("Fail while waiting for down command");
-    let mut bit_rate_commmand = Command::new("sudo")
-        .arg("ip")
-        .arg("link")
-        .arg("set")
-        .arg("can0")
-        .arg("type")
-        .arg("can")
-        .arg("bitrate")
-        .arg("1000000")
-        .spawn()
-        .expect("bit rate command did not work");
-    bit_rate_commmand
-        .wait()
-        .expect("Fail while waiting for bit rate");
-    let mut up_command = Command::new("sudo")
-        .arg("ifconfig")
-        .arg("can0")
-        .arg("up")
-        .spawn()
-        .expect("up command did nto work");
-    up_command
-        .wait()
-        .expect("Fail while waiting for up command");
+    // Define the set of options for the create.
+    // Use an ID for a persistent session.
+    let create_opts = mqtt::CreateOptionsBuilder::new()
+        .server_uri(host)
+        .client_id(DFLT_CLIENT.to_string())
+        .finalize();
 
-    let mut stream = UnixStream::connect(ipc_path).unwrap();
-    let (tx, rx) = channel();
-    //open can socket channel at name can0
-    const CAN_CHANNEL: &str = "can0";
-    let socket = CANSocket::open(&CAN_CHANNEL);
-    let socket = match socket {
-        Ok(socket) => socket,
-        Err(err) => {
-            println!("Failed to open CAN socket: {}", err);
-            return;
+    // Create a client.
+    let mut cli = mqtt::Client::new(create_opts).unwrap_or_else(|err| {
+        println!("Error creating the client: {:?}", err);
+        process::exit(1);
+    });
+
+    // Initialize the consumer before connecting.
+    let rx = cli.start_consuming();
+
+    // Define the set of options for the connection.
+    let lwt = mqtt::MessageBuilder::new()
+        .topic("test")
+        .payload("Consumer lost connection")
+        .finalize();
+    let conn_opts = mqtt::ConnectOptionsBuilder::new()
+        .keep_alive_interval(Duration::from_secs(20))
+        .clean_session(false)
+        .will_message(lwt)
+        .finalize();
+
+    // Connect and wait for it to complete or fail.
+    if let Err(e) = cli.connect(conn_opts) {
+        println!("Unable to connect:\n\t{:?}", e);
+        process::exit(1);
+    }
+
+    // Subscribe topics.
+    subscribe_topics(&cli);
+
+    println!("Processing requests...");
+    for msg in rx.iter() {
+        if let Some(msg) = msg {
+            println!("{}", msg);
         }
-    };
-    thread::spawn(move || loop {
```

Comment:
this shoudlnt be deleted. It should be moved to a seperate file called like NERO Socket